src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs

```diff
@@ -372,10 +372,52 @@ private NamespaceSymbol ResolveExternAliasTarget(BindingDiagnosticBag diagnostic
             return target;
         }
 
-        private NamespaceOrTypeSymbol ResolveAliasTarget(NameSyntax syntax, BindingDiagnosticBag diagnostics, ConsList<TypeSymbol>? basesBeingResolved)
+        private NamespaceOrTypeSymbol ResolveAliasTarget(
+            UsingDirectiveSyntax usingDirective,
+            BindingDiagnosticBag diagnostics,
+            ConsList<TypeSymbol>? basesBeingResolved)
         {
-            var declarationBinder = ContainingSymbol.DeclaringCompilation.GetBinderFactory(syntax.SyntaxTree).GetBinder(syntax).WithAdditionalFlags(BinderFlags.SuppressConstraintChecks | BinderFlags.SuppressObsoleteChecks);
-            return declarationBinder.BindNamespaceOrTypeSymbol(syntax, diagnostics, basesBeingResolved).NamespaceOrTypeSymbol;
+            if (usingDirective.UnsafeKeyword != default)
+            {
+                MessageID.IDS_FeatureUsingTypeAlias.CheckFeatureAvailability(diagnostics, usingDirective, usingDirective.UnsafeKeyword.GetLocation());
+            }
+            else if (usingDirective.Type is not NameSyntax)
+            {
+                MessageID.IDS_FeatureUsingTypeAlias.CheckFeatureAvailability(diagnostics, usingDirective.Type);
+            }
+
+            var syntax = usingDirective.Type;
+            var flags = BinderFlags.SuppressConstraintChecks | BinderFlags.SuppressObsoleteChecks;
+            if (usingDirective.UnsafeKeyword != default)
+            {
+                this.CheckUnsafeModifier(DeclarationModifiers.Unsafe, usingDirective.UnsafeKeyword.GetLocation(), diagnostics);
+                flags |= BinderFlags.UnsafeRegion;
+            }
+
+            var declarationBinder = ContainingSymbol.DeclaringCompilation
+                .GetBinderFactory(syntax.SyntaxTree)
+                .GetBinder(syntax)
+                .WithAdditionalFlags(flags);
+
+            var annotatedNamespaceOrType = declarationBinder.BindNamespaceOrTypeSymbol(syntax, diagnostics, basesBeingResolved);
+
+            // `using X = RefType?;` is not legal.
+            if (usingDirective.Type is NullableTypeSyntax nullableType &&
+                annotatedNamespaceOrType.TypeWithAnnotations.NullableAnnotation == NullableAnnotation.Annotated &&
+                annotatedNamespaceOrType.TypeWithAnnotations.Type?.IsReferenceType is true)
+            {
+                diagnostics.Add(ErrorCode.ERR_BadNullableReferenceTypeInUsingAlias, nullableType.QuestionToken.GetLocation());
+            }
+
+            var namespaceOrType = annotatedNamespaceOrType.NamespaceOrTypeSymbol;
+            if (namespaceOrType is TypeSymbol { IsNativeIntegerWrapperType: true } &&
+                (usingDirective.Type.IsNint || usingDirective.Type.IsNuint))
+            {
+                // using X = nint;
+                MessageID.IDS_FeatureUsingTypeAlias.CheckFeatureAvailability(diagnostics, usingDirective.Type);
```

Comment:
>MessageID.IDS_FeatureUsingTypeAlias.CheckFeatureAvailability(diagnostics, usingDirective.Type); [](http://example.com/codeflow?start=16&length=95)

Are we reporting this error compiling the following with -langversion:11?
```csharp
using A=nint;
class nint { }
```

Reply:
let me check.

Reply:
we are not (which appears to be the correct behavior).  Have added tests to explicitly validate that thsi is so.